### PR TITLE
unit test issue #19

### PR DESCRIPTION
**Pull request by mhmdahmedfathi**
to solve the issue https://github.com/InternetHealthReport/internet-yellow-pages/issues/19
unit_test.py file is in the root directory, you can run it like any other Python file. "python3 unit_test.py"
The script runs all the crawlers and post scripts and sends an additional argument , "unit_test",to each one so they can start testing.  

unit_test.py

- It will call the config file and get all the crawlers and post scripts from the config file, loop over each one, and run them.

In the base crawler, add two methods:

- count_relations: to count the number of relations that the crawler has

- unit_test: count the relations of the crawler, log the number of the relations, and then run the crawler. After it finishes, count the relation again and see if the number of the relation after running is greater or not, and print if the assertion passed or failed.

and for all crawlers added: check if there is an additional arg "unit_test," and if yes, then it runs the unit test method.

for the postscripts

- address_family script:

        -  add one method: unit_test, which runs the script, and after this, see if there is any IP or prefix that is not IPv4 or IPv6, and if yes, it failed and printed an assertion failure; else, it passed.
        - in the main: add a condition to see if there is an additional arg "unit_test," and if yes, run this method.

- ip2prefix.py script

     -  add two methods :
              - count_relation: to count the relation that the IP has
              - unit_test: count the relations before running the test and after running the test and see if the number of relations increases or not, and log and print the result.

     -  in the main program: add a condition to see if there is an additional arg "unit_test," and if yes, run this method.

### CODE REVIEW

File: iyp/crawlers/apnic/eyeball.py [hunk 1]
```diff
@@ -94,5 +94,8 @@ def run(self):
     logging.info("Started: %s" % sys.argv)
 
     apnic = Crawler(ORG, URL, NAME)
-    apnic.run()
-    apnic.close()
+    if len(sys.argv) > 1 and sys.argv[1] == 'unit_test':
+        apnic.unit_test(logging)
+    else :
+        apnic.run()
+        apnic.close()
```
Comment (romain-fontugne) [inline iyp/crawlers/apnic/eyeball.py:101]: The call to close should be outside of the else statement

File: iyp/__init__.py [hunk 2]
```diff
@@ -425,6 +425,27 @@ def fetch(self):
...
+        return result['count']
+    
+    def unit_test(self,logging):
+        relation_count = self.count_relations()
+        logging.info("Relations before starting: %s" % relation_count)
+        self.run()
+        relation_count_new = self.count_relations()
+        logging.info("Relations after starting: %s" % relation_count_new)
+        self.close()
+        #assert relation_count_new > relation_count
```
Comment (romain-fontugne) [inline iyp/__init__.py:444]: I like the assert, so that the test really fails. But I'd put it at the very end of the method, so we can still show the log

**Review by romain-fontugne — CHANGES_REQUESTED** [review]

**Comment by romain-fontugne** [comment]
Thanks, that's very close to be merged. Can you please make sure all the call to the crawler close method are outside of the else statement (I have commented only one but this happened several times)

File: iyp/crawlers/apnic/eyeball.py (hunk 1 above)
Comment (mhmdahmedfathi) [reply iyp/crawlers/apnic/eyeball.py:101]: Unit test method already use close inside it

File: iyp/__init__.py (hunk 2 above)
Comment (mhmdahmedfathi) [reply iyp/__init__.py:444]: So I have to uncomment the assert line but I have to put it in the end after the close

File: iyp/crawlers/apnic/eyeball.py (hunk 1 above)
Comment (romain-fontugne) [reply iyp/crawlers/apnic/eyeball.py:101]: ah, I see. Ok then you can just ignore this!

File: iyp/__init__.py (hunk 2 above)
Comment (romain-fontugne) [reply iyp/__init__.py:444]: yes!

Comment (mhmdahmedfathi) [reply iyp/__init__.py]: I have added it and pushed it, you can see it now

**Review by romain-fontugne — APPROVED** [review]
thanks!

**Pull request merged**